contents/performance/Performance.md

```diff
@@ -34,6 +35,21 @@ If you are interested in learning more about Data Structures and Algorithms, you
 * [MIT online course](http://courses.csail.mit.edu/6.851/spring12/lectures/): A online course by MIT on data structures and algorithms
 * [topcoder.com](https://www.topcoder.com/community/data-science/data-science-tutorials/the-importance-of-algorithms/): A good write up on the importance of algorithms
 
+## Memory Management
+
+Memory management is important for performance optimization for Computer systems. One of the common techniques in [algorithms optimisation](#using-appropriate-data-structures-and-algorithms) is space and time trade off, where we increase runtime memory usage to decrease overall runtime. While this theoretically optimizes your system runtime, it might overall slowdown the system due to [Thrashing](https://en.wikipedia.org/wiki/Thrashing_(computer_science)). Detecting if the performance slowdown is memory related can be done with appropriate [memory profiling](PerformanceProfiling.md).
```

Comment:
Possible to explain `Thrashing` here in short?
It quite doesn't make  sense to link it when the sentence only makes sense if that term is known